src/GridFieldNestedForm.php

```diff
@@ -0,0 +1,517 @@
+use SilverStripe\Forms\GridField\GridField_HTMLProvider;
+use SilverStripe\Forms\GridField\GridField_SaveHandler;
+use SilverStripe\Forms\GridField\GridField_URLHandler;
+use SilverStripe\Forms\GridField\GridFieldStateAware;
+use SilverStripe\ORM\DataList;
+use SilverStripe\ORM\DataObject;
+use SilverStripe\ORM\DataObjectInterface;
+use SilverStripe\ORM\Filterable;
+use SilverStripe\ORM\Hierarchy\Hierarchy;
+use SilverStripe\ORM\SS_List;
+use SilverStripe\Versioned\Versioned;
+use SilverStripe\View\ViewableData;
+use Symbiote\GridFieldExtensions\GridFieldOrderableRows;
+
+/**
+ * Gridfield component for nesting GridFields
+ */
+class GridFieldNestedForm extends AbstractGridFieldComponent implements
+    GridField_URLHandler,
+    GridField_ColumnProvider,
+    GridField_SaveHandler,
+    GridField_HTMLProvider,
+    GridField_DataManipulator
+{
+    use Configurable, GridFieldStateAware;
+    
+    /**
+     * The key used in the post data to identify nested form data
+     */
+    const POST_KEY = 'GridFieldNestedForm';
+
+    private static $allowed_actions = [
+        'handleNestedItem'
+    ];
+
+    /**
+     * The default max nesting level. Nesting further than this will throw an exception.
+     *
+     * @var boolean
+     */
+    private static $default_max_nesting_level = 10;
+    
+    /**
+     * @var string
+     */
+    private $name;
+
+    /**
+     * @var bool
+     */
+    private $expandNested = false;
+
+    /**
+     * @var bool
+     */
+    private $forceCloseNested = false;
+
+    /**
+     * @var GridField
+     */
+    private $gridField = null;
+
+    /**
+     * @var string
+     */
+    private $relationName = 'Children';
+
+    /**
+     * @var bool
+     */
+    private $inlineEditable = false;
+
+    /**
+     * @var callable|string
+     */
+    private $canExpandCheck = null;
+
+    /**
+     * @var int
+     */
+    private $maxNestingLevel = null;
+    
+    public function __construct($name = 'NestedForm')
+    {
+        $this->name = $name;
+    }
+    
+    /**
+     * Get the grid field that this component is attached to
+     */
+    public function getGridField(): GridField
+    {
+        return $this->gridField;
+    }
+
+    /**
+     * Get the relation name to use for the nested grid fields
+     */
+    public function getRelationName(): string
+    {
+        return $this->relationName;
+    }
+    
+    /**
+     * Set the relation name to use for the nested grid fields
+     */
+    public function setRelationName(string $relationName)
+    {
+        $this->relationName = $relationName;
+        return $this;
+    }
+
+    /**
+     * Get whether the nested grid fields should be inline editable
+     */
+    public function getInlineEditable(): bool
+    {
+        return $this->inlineEditable;
+    }
+
+    /**
+     * Set whether the nested grid fields should be inline editable
+     */
+    public function setInlineEditable(bool $editable)
```

Comment:
```suggestion
    public function setInlineEditable(bool $editable): static
```
Please do this for all other `set*` methods which return `$this`